lib/puppet/provider/cs_location/crm.rb

```diff
@@ -81,7 +102,7 @@ def flush
     updated = "location #{@property_hash[:name]} #{@property_hash[:primitive]}"
 
     if feature?(:discovery)
-      updated << " resource-discovery=#{@property_hash[:resource_discovery]}"
+      updated << " resource-discovery=#{@property_hash[:resource_discovery]}" unless @property_hash[:resource_discovery].nil?
```

Comment:
what's the difference between `if feature?(:discovery)` and `unless @property_hash[:resource_discovery].nil?`?

Reply:
where is `@property_hash[:resource_discovery]` set anyway?

Reply:
When one sets parameter resource_discovery in a puppet manifest, but the pacemaker version does not support it yet (hence the feature check), it will complain if the parameter is applied. When pacemaker supports it, the parameter should only be applied if it has a value (hence the unless), empty values are failing - i saw that on earlier (local) beaker tests.
The property(_hash) is set here https://github.com/voxpupuli/puppet-corosync/blob/503a96f8bd133671ef72fb7642dbc90fc23a3ee6/lib/puppet/provider/cs_location/crm.rb#L84